Mage.Sets/src/mage/cards/s/SharedFate.java

```diff
@@ -151,19 +152,61 @@ public SharedFatePlayEffect copy() {
 
     @Override
     public boolean applies(UUID objectId, Ability source, UUID affectedControllerId, Game game) {
-        Permanent sourcePermanent = game.getPermanent(source.getSourceId());
-        if (sourcePermanent != null) {
-            ExileZone exileZone = game.getExile().getExileZone(CardUtil.getExileZoneId(source.getSourceId().toString() + sourcePermanent.getZoneChangeCounter(game) + affectedControllerId.toString(), game));
-            if (exileZone != null) {
-                Card card = exileZone.get(objectId, game);
-                Player player = game.getPlayer(affectedControllerId);
-                if (card != null && player != null) {
-                    player.lookAtCards(card.getName(), card, game);
-                    // You already get asked to confirm when casting a spell, but not when playing a land.
-                    return !card.getCardType().contains(CardType.LAND) || player.chooseUse(Outcome.Benefit, "Play " + card.getName() + "?", source, game);
+       if (affectedControllerId.equals(source.getControllerId()) && game.getState().getZone(objectId).equals(Zone.EXILED)) {
```

Comment:
Why are you checking whether the affected controller is the source controller? It looks like Shared Fate is supposed to work for all players.